test/bowling.test.js

```diff
@@ -0,0 +1,187 @@
+  })
+  it('Will add up the scores when asked to', () => {
+    bowling = new Bowling()
+    bowling.myRoll(6)
+    bowling.myRoll(3)
+    bowling.myRoll(3)
+    expect(bowling.calculateBasic()).toEqual(12);
+  })
+
+  it('Will add up strikes and spares correctly', () => {
+    bowling = new Bowling()
+    bowling.myRoll(10)
+    bowling.myRoll(3)
+    expect(bowling.calculateStrikes()).toEqual(3)
+    bowling2 = new Bowling()
+    bowling2.myRoll(7)
+    bowling2.myRoll(3)
+    bowling2.myRoll(4)
+    expect(bowling2.calculateStrikes()).toEqual(4)
+    bowling3 = new Bowling()
+    bowling3.myRoll(10)
+    bowling3.myRoll(10)
+    bowling3.myRoll(4)
+    expect(bowling3.calculateStrikes()).toEqual(18)
+  })
+
+  it('Can fully add up the current score', () => {
+    bowling = new Bowling()
+    bowling.myRoll(10)
+    bowling.myRoll(3)
+    bowling.myRoll(7)
+    bowling.myRoll(4)
+    expect(bowling.totalScore()).toEqual(38)
+  })
+
+  it('Will not let you bowl past 10 + bonus', () => {
+    bowling = new Bowling()
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(5)
+    bowling.myRoll(4)
+    expect(bowling.myRoll(1)).toEqual("The game is over! No more rolls")
+  })
+
+  it('Will not let you bowl past 10 if you do not get the bonus', () => {
+    bowling = new Bowling()
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    expect(bowling.myRoll(1)).toEqual("The game is over! No more rolls")
+  })
+
+
+  it ('Feature test 1', () => {
+    bowling = new Bowling()
+    bowling.myRoll(5)
+    bowling.myRoll(4)
+    bowling.myRoll(7)
+    bowling.myRoll(2)
+    bowling.myRoll(8)
+    bowling.myRoll(2)
+    bowling.myRoll(10)
+    bowling.myRoll(7)
+    bowling.myRoll(2)
+    bowling.myRoll(8)
+    bowling.myRoll(1)
+    bowling.myRoll(6)
+    bowling.myRoll(4)
+    bowling.myRoll(9)
+    bowling.myRoll(1)
+    bowling.myRoll(5)
+    bowling.myRoll(3)
+    bowling.myRoll(3)
+    bowling.myRoll(5)
+    expect(bowling.totalScore()).toEqual(125)
+  })
+
+  it ('Feature test 2 - perfect game', () => {
+    bowling = new Bowling()
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    bowling.myRoll(10)
+    expect(bowling.myRoll(10)).toEqual({"1": [10], "10": [10, 10, 10], "2": [10], "3": [10], "4": [10], "5": [10], "6": [10], "7": [10], "8": [10], "9": [10]})
+    expect(bowling.totalScore()).toEqual(300)
+  })
+
```

Comment:
I'm a bit confused how you get 300 if in the 10th frame you cannot add your full result ... but you only add 10. But well done in adding the "perfect game"